neo4j/internal/bolt/bolt5.go

```diff
@@ -324,14 +325,16 @@ func (b *bolt5) TxBegin(
 	}
 
 	b.queue.appendBegin(tx.toMeta(b.log, b.logId), b.beginResponseHandler())
-	if b.queue.send(ctx); b.err != nil {
-		return 0, b.err
-	}
-	if err := b.queue.receiveAll(ctx); err != nil {
-		return 0, err
-	}
-	if b.err != nil { // onNextMessageErr kicked in
-		return 0, b.err
+	if syncMessages {
+		if b.queue.send(ctx); b.err != nil {
+			return 0, b.err
+		}
+		if err := b.queue.receiveAll(ctx); err != nil {
+			return 0, err
+		}
+		if b.err != nil { // onNextMessageErr kicked in
+			return 0, b.err
+		}
 	}
 
 	b.state = bolt5Tx
```

Comment:
we are not actually in tx state on the server so is setting `b.state` to `bolt5Tx` an issue? likewise in bolt4

Reply:
Not only is it not an issue, it's necessary. Further down down in `bolt5.run` you'll see
```go
	if b.state == bolt5Ready {
		b.state = bolt5Streaming
	} else if b.state == bolt5Tx {
		b.state = bolt5StreamingTx
	}
	return stream, nil
```

That would put the connection if the wrong state on the pipelined `RUN` if `begin` weren't setting `state` to `bolt5Tx`.